### PR TITLE
DrivAerML: metric-aware w=0.03 + lr=4e-4 (low-weight low-lr)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Explore a lower learning rate (4e-4 vs 5e-4) combined with an intermediate metric-aware weight (w=0.03). Lower LR may allow more stable integration of the rel-L2 auxiliary loss, potentially finding a better optimum. The combination of reduced gradient noise (lower LR) + moderate metric alignment (w=0.03) is unexplored.

## Current DM Baseline (PR #3302)

- **val_primary/surface_rel_l2_pct:** 3.700% (Trial B, w=0.05, seed=42) — **BREAKS BELOW AB-UPT SOTA (3.71%)**
- **test_primary/surface_rel_l2_pct:** 4.500% batch-limited (full-eval pending)
- **Config:** 4L/512d/8H, AdamW **lr=5e-4**, T_max=30, EMA=0.9995, gc=0.5, `--dm-loss mse_plus_raw_rel_l2 --dm-rel-l2-weight 0.05`

## Instructions to the Student

Run the following command:

```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml \
  --dm-loss mse_plus_raw_rel_l2 \
  --dm-rel-l2-weight 0.03 \
  --optimizer adamw \
  --lr 4e-4 \
  --cosine-t-max 30 \
  --grad-clip 0.5 \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --ema-decay 0.9995 \
  --ema-mode fixed \
  --no-compile-model \
  --save-checkpoint \
  --eval-interval 10 \
  --seed 42 \
  --wandb-group dm-metric-aware-sweep \
  --kill-if-best-val-above 150:8.0 \
  --kill-if-no-improvement 200:0.05
```

**What to report:**
1. Best `val_primary/surface_rel_l2_pct` and the epoch it occurred
2. Best `test_primary/surface_rel_l2_pct` (batch-limited) and the epoch
3. W&B run ID
4. Convergence speed — does lr=4e-4 converge slower? Does it reach a lower floor?
5. Compare to w=0.03 at lr=5e-4 (einar #3323) if available

**Target:** val < 3.700%, test < 4.320%

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT sanji:

## Results

### DrivAerML: metric-aware w=0.03 + lr=4e-4 (seed=42)

| Metric | This Run | Baseline (PR #3302, w=0.05, lr=5e-4) | Target |
|--------|----------|---------------------------------------|--------|
| **val_primary/surface_rel_l2_pct** | **3.967%** (ep 724) | **3.700%** (ep 780) | < 3.700% |
| **test_primary/surface_rel_l2_pct** | **4.489%** (batch-limited) | **4.320%** (Trial A) / 4.500% (Trial B) | < 4.320% |

**W&B run:** `7hrayanv` — https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/7hrayanv
**W&B group:** `dm-metric-aware-sweep`

### Command used

```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml \
  --dm-loss mse_plus_raw_rel_l2 \
  --dm-rel-l2-weight 0.03 \
  --optimizer adamw \
  --lr 4e-4 \
  --cosine-t-max 30 \
  --grad-clip 0.5 \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --ema-decay 0.9995 \
  --ema-mode fixed \
  --no-compile-model \
  --save-checkpoint \
  --eval-interval 10 \
  --seed 42 \
  --agent sanji \
  --wandb-name "sanji/dm-metric-aware-w003-lr4e4" \
  --wandb-group dm-metric-aware-sweep \
  --kill-if-best-val-above 150:8.0 \
  --kill-if-no-improvement 200:0.05
```

### Peak memory
~7.4 GB VRAM on a single RTX PRO 6000 Blackwell (97 GB available)

### What happened

**The hypothesis did NOT beat the baseline.** Val reached 3.967% vs the 3.700% target, and test was 4.489% vs the 4.320% target.

**Convergence dynamics:**
- The lower LR (4e-4 vs 5e-4) converged more slowly as expected — the model was still finding improvements deep into training (epoch 724)
- The convergence curve showed a characteristic staircase pattern: periods of plateau (e.g. epochs 320-360, 430-550, 620-710) punctuated by breakthroughs
- The final breakthrough from 4.01% → 3.97% came at the very end (epoch 710-724) before the timeout hit
- With more time, the model *might* have continued improving — but the gap to 3.70% is significant (0.27pp)

**Comparison to w=0.03 at lr=5e-4 (einar #3323, if available):**
- This run's best val (3.97%) can be compared when einar's results are in. The hypothesis that lower LR + w=0.03 finds a better optimum was not confirmed — the baseline's stronger metric alignment (w=0.05) at higher LR (5e-4) still wins.

**Analysis:**
- w=0.03 provides too little metric-aware signal compared to w=0.05 — the rel-L2 contribution was only ~0.1-0.3% of total loss throughout training
- The lower LR slowed convergence but did not find a deeper basin than the baseline's w=0.05 configuration
- The combination suggests that metric alignment weight matters more than LR stability for this task

### Suggested follow-ups
- Try w=0.05 + lr=4e-4 to isolate whether the lower LR helps when metric alignment is stronger
- The staircase convergence pattern suggests this model could benefit from longer training at optimal weight — try w=0.05 with extended epochs
- Consider adaptive metric-aware weight scheduling (start low, ramp up)

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3329 because ADVISOR: Closing — val=3.967% at T_max=30, well below both the metric-aware baseline (3.700%) and the T_max=36 frontier (3.622%). Lower LR (4e-4) + lower weight (0.03) converges slower without finding a deeper basin. The T_max=30 metric-aware weight curve is well-characterized: w=0.05 at lr=5e-4 is the optimum. All effort now shifts to T_max=36 compounds..
